crates/sqlite_store/src/store.rs

```diff
@@ -0,0 +1,827 @@
+use bdk_chain::bitcoin::consensus::{deserialize, serialize};
+use bdk_chain::bitcoin::hashes::Hash;
+use bdk_chain::bitcoin::{Amount, Network, OutPoint, ScriptBuf, Transaction, TxOut};
+use bdk_chain::bitcoin::{BlockHash, Txid};
+use bdk_chain::miniscript::descriptor::{Descriptor, DescriptorPublicKey};
+use rusqlite::{named_params, Connection};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+use std::marker::PhantomData;
+use std::path::Path;
+use std::str::FromStr;
+use std::sync::{Arc, Mutex};
+
+use crate::schema::Schema;
+use crate::{ChangeSet, Error};
+use bdk_chain::{
+    indexed_tx_graph, keychain, local_chain, tx_graph, Anchor, Append, DescriptorExt, DescriptorId,
+};
+
+/// Persists [`ChangeSet`] data in to a relational schema based SQLite database file.
+///
+/// The changesets loaded or stored represent changes to keychain and blockchain data. If the
+/// keychain (K) is a simple enum without variant fields you must enable the
+/// [serde internal tag](https://serde.rs/enum-representations.html#internally-tagged) feature.
+#[derive(Debug)]
+pub struct Store<K: Send, A: Send> {
+    // A rusqlite connection to the SQLite database. Uses a Mutex for thread safety.
+    conn: Mutex<Connection>,
+    keychain_marker: PhantomData<K>,
+    anchor_marker: PhantomData<A>,
+}
+
+impl<K, A> Store<K, A>
+where
+    K: Ord + for<'de> Deserialize<'de> + Serialize + Send,
+    A: Anchor + for<'de> Deserialize<'de> + Serialize + Send,
+{
+    /// Creates a new store from a [`Path`].
+    ///
+    /// The file must be able to be opened with read and write permissions.
+    pub fn new<P: AsRef<Path>>(path: P) -> Result<Self, rusqlite::Error> {
```

Comment:
I think we should let the caller construct the `Connection` and we have `Connection` as the input. Not only could we remove the `new_memory` method, but also allows the caller to set custom flags. Or maybe there is an argument saying that this is dangerous in some way?

Reply:
I prefer keeping things simple and `new(path)` is the only one the users would generally need. For `new_memory()` as described in the docs it's primarily for testing. If we find a case where a user needs to tweek the flags then we can add a `new_with_flags(path,flags)` but for now I can't think of any reason they'd need to.

Reply:
Honestly, I feel like the current implementation is sacrificing a lot of flexibility for very little extra convenience.